### PR TITLE
std: Clean out #[deprecated] APIs

### PR DESCRIPTION
**Pull request by alexcrichton**
This commit cleans out a large amount of deprecated APIs from the standard
library and some of the facade crates as well, updating all users in the
compiler and in tests as it goes along.

### CODE REVIEW

File: src/libcollections/vec.rs [hunk 1]
```diff
@@ -423,15 +423,6 @@ impl<T> Vec<T> {
         }
     }
 
-    /// Deprecated: use `&mut s[..]` instead.
-    #[inline]
-    #[unstable(feature = "collections",
-               reason = "will be replaced by slice syntax")]
-    #[deprecated(since = "1.0.0", reason = "use &mut s[..] instead")]
-    pub fn as_mut_slice(&mut self) -> &mut [T] {
```
Comment (aturon) [inline src/libcollections/vec.rs]: May as well leave this around, it's coming back :)

Comment (alexcrichton) [inline src/libcollections/vec.rs]: Re-added

**Comment by aturon** [comment]
r=me after a rebase.

Thanks for tirelessly doing these! Spring cleaning :)

**Comment by alexcrichton** [comment]
Merged in #23936

**Pull request closed without merging**